### ISSUE TITLE
这个错误是怎么回事？

### ISSUE THREAD

**Issue by mingxing0769** [KittenCN/stock_prediction#22]
运行：python predict.py后
全屏闪：
code: data_queue, train error: _foreach_addcdiv_() received an invalid combination of arguments - got (list, list, tuple, list), but expected one of: [07:40<54:07:02, 15.94it/s]
 * (tuple of Tensors self, tuple of Tensors tensor1, tuple of Tensors tensor2, tuple of Scalars scalars)
      didn't match because some of the arguments have invalid types: (list of [Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter, Parameter

**Comment by KittenCN** [comment]
修改了什么东西吗？

**Comment by mingxing0769** [comment]
> 修改了什么东西吗？

并没有修改什么，按说明下载了数据 和给数据做了预处理，后运行:python predict.py  没加其他参数  加其他参数也会提示data_queue 传输给_foreach_addcdiv_() 数据类型不对

**Comment by jiangfubang** [comment]
我也是

**Comment by jiangfubang** [comment]
不知道是不是python版本问题或者模块版本问题，可以把你的python版本和模块版本号发一下吗

**Comment by mingxing0769** [comment]
Python 3.11.3 
CUDA 12.1
Package                      Version
---------------------------- ------------------------
absl-py                      1.4.0
akshare                      1.9.93
appdirs                      1.4.4
astunparse                   1.6.3
beautifulsoup4               4.12.2
bs4                          0.0.1
cachetools                   5.3.0
certifi                      2023.5.7
cffi                         1.15.1
chardet                      3.0.4
charset-normalizer           3.1.0
click                        8.1.3
colorama                     0.4.6
comtypes                     1.1.14
contourpy                    1.0.7
cryptography                 40.0.2
cycler                       0.11.0
decorator                    5.1.1
dill                         0.3.6
et-xmlfile                   1.1.0
filelock                     3.12.0
flatbuffers                  23.5.9
fonttools                    4.39.4
frozendict                   2.3.8
fsspec                       2023.5.0
gast                         0.4.0
google-auth                  2.18.1
google-auth-oauthlib         1.0.0
google-pasta                 0.2.0
googletrans                  3.0.0
grpcio                       1.54.2
h11                          0.9.0
h2                           3.2.0
h5py                         3.8.0
hpack                        3.0.0
hstspreload                  2023.1.1
html5lib                     1.1
httpcore                     0.9.1
httpx                        0.13.3
huggingface-hub              0.14.1
hyperframe                   5.2.0
idna                         2.10
jax                          0.4.10
Jinja2                       3.1.2
joblib                       1.2.0
jsonpath                     0.82
keras                        2.12.0
kiwisolver                   1.4.4
libclang                     16.0.0
lxml                         4.9.2
Markdown                     3.4.3
MarkupSafe                   2.1.2
matplotlib                   3.7.1
ml-dtypes                    0.1.0
mplfinance                   0.12.9b7
mpmath                       1.3.0
multitasking                 0.0.11
networkx                     3.1
numpy                        1.23.5
oauthlib                     3.2.2
opencv-python                4.7.0.72
openpyxl                     3.1.2
opt-einsum                   3.3.0
packaging                    23.1
pandas                       2.0.1
Pillow                       9.5.0
pip                          23.1.2
plyer                        2.1.0
pocketsphinx                 5.0.1
prefetch-generator           1.0.3
protobuf                     4.23.1
py-mini-racer                0.6.0
pyasn1                       0.5.0
pyasn1-modules               0.3.0
PyAudio                      0.2.13
pycparser                    2.21
pyparsing                    3.0.9
pypinyin                     0.49.0
pypiwin32                    223
pytesseract                  0.3.10
python-dateutil              2.8.2
pyttsx3                      2.90
pytz                         2023.3
pywin32                      306
PyYAML                       6.0
regex                        2023.5.5
requests                     2.31.0
requests-oauthlib            1.3.1
rfc3986                      1.5.0
rsa                          4.9
sacremoses                   0.0.53
scikit-learn                 1.2.2
scipy                        1.10.1
sentencepiece                0.1.98
setuptools                   65.5.0
simplejson                   3.19.1
six                          1.16.0
sniffio                      1.3.0
sounddevice                  0.4.6
soundfile                    0.12.1
soupsieve                    2.4.1
SpeechRecognition            3.10.0
sympy                        1.12
tabulate                     0.9.0
tensorboard                  2.12.3
tensorboard-data-server      0.7.0
tensorflow                   2.12.0
tensorflow-estimator         2.12.0
tensorflow-intel             2.12.0
tensorflow-io-gcs-filesystem 0.31.0
termcolor                    2.3.0
threadpoolctl                3.1.0
tokenizers                   0.13.3
torch                        2.1.0.dev20230523+cu121
torchaudio                   2.1.0.dev20230524+cu121
torchvision                  0.16.0.dev20230524+cu121
tqdm                         4.65.0
transformers                 4.29.2
tushare                      1.2.89
typing_extensions            4.6.1
tzdata                       2023.3
urllib3                      1.26.16
webencodings                 0.5.1
websocket-client             0.57.0
Werkzeug                     2.3.4
wheel                        0.40.0
wrapt                        1.14.1
xlrd                         2.0.1
yfinance                     0.2.18

**Comment by jiangfubang** [comment]
@mingxing0769 应该是咱俩cuda版本太新了

**Comment by KittenCN** [comment]
![image](https://github.com/KittenCN/stock_prediction/assets/10828489/f3875659-c747-4b3a-85b5-e4dd3bd14f0a)
我目前只在11.5和12.0上测试过。。
![image](https://github.com/KittenCN/stock_prediction/assets/10828489/2f112fb2-5c9b-497a-9575-469b7e5e7333)
如果不修改东西，训练起来应该没有问题的

**Comment by KittenCN** [comment]
其他文件的版本requirements文件有说明

**Comment by jiangfubang** [comment]
cuda12.1不行，我换成11.6可以了

**Comment by mingxing0769** [comment]
> 

请问torch 是哪个版本的？

**Comment by mingxing0769** [comment]
> > cuda12.1不行，我换成11.6可以了
> 
> 估计就是版本原因了，我没有条件测试12.0以上的版本

cuda版本11.5还是不行 python版本是311 torch              2.0.1+cu117
torchaudio         2.0.2+cu117
torchvision        0.15.2+cu117

**Comment by mingxing0769** [comment]
终于找到原因了torch版本的问题 我用的python 版本太高了点  把python安装3.10 之后 torch版本用1.30 就可以了。
